step.go

```diff
@@ -36,17 +55,105 @@ func (a ArtifactPull) ProcessConfig() (Config, error) {
 
 	stepconf.Print(input)
 
+	finishedStages := input.FinishedStages
+	var finishedStagesModel model.FinishedStages
+	if finishedStages != "" {
+		if err := json.Unmarshal([]byte(finishedStages), &finishedStagesModel); err != nil {
+			return Config{}, fmt.Errorf("failed to parse step inputs: %w", err)
+		}
+	}
+
+	appSlug := a.envRepository.Get("BITRISE_APP_SLUG")
+	if appSlug == "" {
+		return Config{}, fmt.Errorf("app slug (BITRISE_APP_SLUG env var) not found")
+	}
+
 	return Config{
-		Example: input.Example,
+		VerboseLogging:        input.Verbose,
+		ArtifactSources:       strings.Split(input.ArtifactSources, ","),
+		FinishedStages:        finishedStagesModel,
+		BitriseAPIAccessToken: string(input.BitriseAPIAccessToken),
+		BitriseAPIBaseURL:     input.BitriseAPIBaseURL,
+		AppSlug:               appSlug,
 	}, nil
 }
 
 func (a ArtifactPull) Run(cfg Config) (Result, error) {
-	// TODO
-	return Result{}, nil
+	a.logger.EnableDebugLog(cfg.VerboseLogging)
+	buildIdGetter := NewBuildIDGetter(cfg.FinishedStages, cfg.ArtifactSources)
+	buildIDs, err := buildIdGetter.GetBuildIDs()
+	if err != nil {
+		return Result{}, err
+	}
+
+	a.logger.Debugf("Downloading artifacts for builds %+v", buildIDs)
+
+	a.logger.Printf("getting the list of artifacts of %d builds", len(buildIDs))
```

Comment:
Log messages seem to have randomly capital or lowercase first character. We usually capitalize I think.